superset-frontend/packages/superset-ui-core/src/color/CategoricalColorScale.ts

```diff
@@ -86,6 +86,12 @@ class CategoricalColorScale extends ExtensibleFunction {
         updatedRange.splice(index, 1);
       }
     });
+    Object.entries(this.parentForcedColors).forEach(([key, value]) => {
+      if (key !== cleanedValue) {
+        const index = updatedRange.indexOf(value);
+        updatedRange.splice(index, 1);
```

Comment:
could you use a filter method on updatedRange here instead? That may be a little more succinct.

Reply:
Yeah, this looks great!

Reply:
Additional suggestion: https://github.com/apache/superset/pull/24670#discussion_r1262907425